src/Actions/DeleteCategory.php

```diff
@@ -15,6 +18,22 @@ public function __construct(Category $category)
 
     protected function transact()
     {
-        return $this->category->delete();
+        $categoryIdsToDelete = [];
+        $threadIdsToDelete = [];
+        if (! $this->category->isEmpty())
+        {
+            $descendantIds = $this->category->descendants->pluck('id')->toArray();
+            $categoryIdsToDelete += $descendantIds;
+            $threadIdsToDelete += Thread::whereIn('category_id', $descendantIds)->withTrashed()->pluck('id')->toArray();
+        }
+
+        $categoryIdsToDelete[] = $this->category->id;
+        $threadIdsToDelete = array_merge($threadIdsToDelete, $this->category->threads()->withTrashed()->pluck('id')->toArray());
```

Comment:
In one place you're doing `+=` and another you're doing `array_merge`? I'd stick to the same method unless there's a particular reason you need to use different ones.

Reply:
They behave slightly differently; `+=` won't overwrite existing keys, whereas `array_merge()` will append values without checking keys (overwriting keys if they exist).

For consistency's sake I'll use `array_merge` in both cases.